### PR TITLE
feature/AKD-142: Added 'Azure Active Directory (AD) Authentication Enabled' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -391,6 +391,7 @@ module.exports = {
         'approvedVmImage'               : require(__dirname + '/plugins/azure/virtualmachines/approvedVmImage.js'),
         'autoOsUpgradesEnabled'         : require(__dirname + '/plugins/azure/virtualmachines/autoOsUpgradesEnabled.js'),
         'noUnattachedDisks'             : require(__dirname + '/plugins/azure/virtualmachines/noUnattachedDisks.js'),
+        'adAuthenticationEnabled'       : require(__dirname + '/plugins/azure/virtualmachines/adAuthenticationEnabled.js'),
```
Comment (AkhtarAmir) [inline exports.js:394]:
```suggestion
        'vmAdAuthenticationEnabled'     : require(__dirname + '/plugins/azure/virtualmachines/vmAdAuthenticationEnabled.js'),
```

File: plugins/azure/virtualmachines/adAuthenticationEnabled.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+var async = require('async');
+
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'Azure Active Directory (AD) Authentication Enabled',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/adAuthenticationEnabled.js:6]:
```suggestion
    title: 'VM Active Directory (AD) Authentication Enabled',
```

File: plugins/azure/virtualmachines/adAuthenticationEnabled.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    category: 'Virtual Machines',
+    description: 'Ensures that Azure Active Directory (AD) authentication is enabled of virtual machines.',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/adAuthenticationEnabled.js:8]:
```suggestion
    description: 'Ensures that Azure Active Directory (AD) authentication is enabled for virtual machines.',
```

File: plugins/azure/virtualmachines/adAuthenticationEnabled.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    virtualMachine.storageProfile.imageReference &&
+                    virtualMachine.storageProfile.imageReference.offer &&
+                    virtualMachine.storageProfile.imageReference.offer.toLowerCase().indexOf('windowsserver') > -1) {
+                    windowsImg = true;
+                } else if (virtualMachine.storageProfile &&
+                    virtualMachine.storageProfile.osDisk &&
+                    virtualMachine.storageProfile.osDisk.osType &&
+                    virtualMachine.storageProfile.osDisk.osType.toLowerCase().indexOf('windows') > -1) {
+                    windowsImg = true;
+                }
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/adAuthenticationEnabled.js:60]: Make this one check

File: plugins/azure/virtualmachines/adAuthenticationEnabled.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+                const adEnabled = virtualMachineExtensions.data.some((virtualMachineExtension) => ((windowsImg && virtualMachineExtension.name && virtualMachineExtension.name === 'AADLoginForWindows') ||
+                        (!windowsImg && virtualMachineExtension.name && virtualMachineExtension.name === 'AADLoginForLinux') ||
+                        (!windowsImg && virtualMachineExtension.name && virtualMachineExtension.name === 'AADSSHLoginForLinux')));
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/adAuthenticationEnabled.js:64]: Make this find or for loop

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/virtualmachines/adAuthenticationEnabled.js (hunk 5 above)
Comment (ali-imran7) [reply plugins/azure/virtualmachines/adAuthenticationEnabled.js:64]: Lets discuss this

**Review by giorod3 — APPROVED** [review]

**Pull request merged**